### PR TITLE
It solves a broken link on HOC doc

### PR DESCRIPTION
**Pull request by Lukyhenson**
## Description

This PR was opened to change the link of `Redux connect` inside the HOC doc.
 
It fix this issue: #1660.

### CODE REVIEW

**Comment by sobrinho** [comment]
@Lukyhenson merge conflict

**Comment by Lukyhenson** [comment]
> @Lukyhenson merge conflict

I'm going to solve that, thanks

File: content/docs/higher-order-components.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ const EnhancedComponent = higherOrderComponent(WrappedComponent);
 
 Whereas a component transforms props into UI, a higher-order component transforms a component into another component.
 
-HOCs are common in third-party React libraries, such as Redux's [`connect`](https://github.com/reactjs/react-redux/blob/master/docs/api/connect.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) and Relay's [`createFragmentContainer`](http://facebook.github.io/relay/docs/en/fragment-container.html).
+HOCs are common in third-party React libraries, such as Redux's [`connect`](https://github.com/reduxjs/react-redux/blob/5.x/docs/api.md#connect) and Relay's [`createFragmentContainer`](http://facebook.github.io/relay/docs/en/fragment-container.html).
```
Comment (lex111) [inline content/docs/higher-order-components.md]: Why not use this link (on the `master` branch)?

https://github.com/reduxjs/react-redux/blob/master/docs/api/connect.md#connect

Comment (Lukyhenson) [reply content/docs/higher-order-components.md]: looks good to me.

**Comment by lex111** [comment]
@Lukyhenson thank you!

**Pull request merged**